### PR TITLE
improve consent validation with joi

### PR DESCRIPTION
**Pull request by doramatadora**
- better validation, with joi
- export validation functions and schema
- keep original test coverage

### CODE REVIEW

File: src/services/validation/consent-api.ts [hunk 1]
```diff
@@ -1,30 +1,54 @@
+import * as Joi from 'joi';
```
Comment (jonnywyatt) [inline src/services/validation/consent-api.ts]: Is this file being exercised by any tests? If not could a couple get added?

File: test/constants.js [hunk 2]
```diff
@@ -1,39 +1,39 @@
 module.exports = {
-    test: {
-        uuid: 'test-uuid',
-        source: 'test-source',
-        scope: 'TEST',
-        category: 'marketing',
-        channel: 'byEmail'
-    },
+	test: {
```
Comment (jonnywyatt) [inline test/constants.js]: Is there a tslint setting for this?

File: test/getUser.spec.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { expect } from 'chai';
 import { getUserBySession, getUserIdAndSessionData } from '../src/read/getUser';
-import {graphQlUserBySession, userIdBySession} from './nocks';
+import { graphQlUserBySession, userIdBySession } from './nocks';
```
Comment (jonnywyatt) [inline test/getUser.spec.ts]: Can this go in tslint?

File: test/nocks.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
-const nock = require('nock');
-const { test, testEnv, responses } = require('./constants');
+const nock = require("nock");
```
Comment (jonnywyatt) [inline test/nocks.js]: Can this go in tslint?

File: src/services/validation/consent-api.ts (hunk 1 above)
Comment (doramatadora) [reply src/services/validation/consent-api.ts]: Yes, ample validation coverage in the UserConsent tests so splitting out would be duplication

File: test/constants.js (hunk 2 above)
Comment (doramatadora) [reply test/constants.js]: This is just standard n-gage with lintspaces running on prepush

File: test/getUser.spec.ts (hunk 3 above)
Comment (doramatadora) [reply test/getUser.spec.ts]: There's no option for it, I was just being pedantic

**Pull request merged**